drivers/usb/udc/udc_dwc2.c

```diff
@@ -1180,6 +1185,10 @@ static int udc_dwc2_ep_enable(const struct device *dev,
 		}
 	}
 
+	dxepctl &= ~USB_DWC2_DEPCTL_STALL;
+	if (dxepctl & USB_DWC2_DEPCTL_NAKSTS) {
+		dxepctl |= USB_DWC2_DEPCTL_CNAK;
+	}
```

Comment:
Maybe add comment
```c
	dxepctl &= ~USB_DWC2_DEPCTL_STALL;
	if (dxepctl & USB_DWC2_DEPCTL_NAKSTS) {
		/* Clear NAK */
		dxepctl |= USB_DWC2_DEPCTL_CNAK;
	}

	dxepctl |= USB_DWC2_DEPCTL_USBACTEP;
```

Reply:
This might not be proper after all. I think we want the endpoint to accept data only after buffer is enqueued. I'll take another look at it.

Reply:
Actually, it is OK - caught myself again on what endpoint "enable" means. The endpoint will NAK if EPENA is not set even if there is room in the RxFIFO. We only set EPENA inside `dwc2_prep_rx()` when we have buffer enqueued.

Reply:
Removed it all together because it was just a workaround to missing `udc_ep_set_busy(dev, cfg->addr, false);` in endpoint dequeue. Now it is working correctly.